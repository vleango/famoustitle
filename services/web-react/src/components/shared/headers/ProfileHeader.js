import React, {Fragment} from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem } from 'reactstrap';

import { startLogout } from '../../../actions/auth';

export class ProfileHeader extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      isOpen: false
    };
  }
  toggle = (e) => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  onLogout = (e) => {
    this.props.startLogout()
  }

  render() {
    return (
      <Navbar color="faded" light expand="md">
        <NavbarBrand tag={Link} to="/">TechWriter</NavbarBrand>
        <NavbarToggler onClick={this.toggle} />
        <Collapse isOpen={this.state.isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <UncontrolledDropdown nav inNavbar>

              { this.props.isAuthenticated ? (
                <Fragment>
                  <DropdownToggle nav caret>
                    { `Hi, ${this.props.firstName}` }
                  </DropdownToggle>
                  <DropdownMenu right>
                    <DropdownItem>
                      Profile
                    </DropdownItem>
                    <DropdownItem>
                      Articles
                    </DropdownItem>
                    <DropdownItem divider />
                    <DropdownItem onClick={this.onLogout}>
                      Logout
                    </DropdownItem>
                  </DropdownMenu>
                </Fragment>

              ) : (
                <Fragment>
                  <DropdownToggle nav caret>
                    Account
                  </DropdownToggle>
                  <DropdownMenu right>
                    <DropdownItem tag={Link} to={'/login'}>
                      Login
                    </DropdownItem>
                    <DropdownItem>
                      Register
                    </DropdownItem>
                  </DropdownMenu>
                </Fragment>
              ) }

            </UncontrolledDropdown>
          </Nav>
        </Collapse>
      </Navbar>
    );
  }

}

const mapStateToProps = (state) => ({
  isAuthenticated: !!state.auth.token,
  firstName: state.auth.firstName,
  lastName: state.auth.lastName
});

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(mapStateToProps, mapDispatchToProps)(ProfileHeader);
