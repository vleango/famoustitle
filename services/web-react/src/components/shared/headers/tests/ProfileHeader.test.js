import React from 'react';
import { shallow } from 'enzyme';

import { ProfileHeader } from '../ProfileHeader';

describe('Shared', () => {
  describe('Headers', () => {
    describe('ProfileHeader', () => {

      let wrapper = shallow(<ProfileHeader />);

      describe('Snapshot', () => {
        it('should correctly render ProfileHeader', () => {
          expect(wrapper).toMatchSnapshot();
        });
      });

      describe('toggle', () => {
        it('should toggle isOpen', () => {
          wrapper.instance().toggle();
          expect(wrapper.state('isOpen')).toEqual(true);
          wrapper.instance().toggle();
          expect(wrapper.state('isOpen')).toEqual(false);
        });
      });

      describe('onLogout', () => {
        const logout = jest.fn();
        wrapper = shallow(<ProfileHeader startLogout={logout} />);
        wrapper.instance().onLogout();
        expect(logout).toHaveBeenCalled();
      });
    });

  });
});
