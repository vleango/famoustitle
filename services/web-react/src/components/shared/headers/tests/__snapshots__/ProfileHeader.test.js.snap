// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Headers ProfileHeader Snapshot should correctly render ProfileHeader 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileHeader
    startLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavbarBrand
          tag={[Function]}
          to="/"
>
          famoustitle
</NavbarBrand>,
        <NavbarToggler
          onClick={[Function]}
          tag="button"
          type="button"
/>,
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
>
          <Nav
                    className="ml-auto"
                    navbar={true}
                    tag="ul"
                    vertical={false}
          >
                    <UncontrolledDropdown
                              inNavbar={true}
                              nav={true}
                    >
                              <Unknown>
                                        <DropdownToggle
                                                  aria-haspopup={true}
                                                  caret={true}
                                                  color="secondary"
                                                  nav={true}
                                        >
                                                  Account
                                        </DropdownToggle>
                                        <DropdownMenu
                                                  flip={true}
                                                  right={true}
                                                  tag="div"
                                        >
                                                  <DropdownItem
                                                            tag={[Function]}
                                                            to="/login"
                                                            toggle={true}
                                                  >
                                                            Login
                                                  </DropdownItem>
                                        </DropdownMenu>
                              </Unknown>
                    </UncontrolledDropdown>
          </Nav>
</Collapse>,
      ],
      "color": "faded",
      "expand": "md",
      "light": true,
      "tag": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "famoustitle",
          "tag": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "famoustitle",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "tag": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
>
            <UncontrolledDropdown
                        inNavbar={true}
                        nav={true}
            >
                        <Unknown>
                                    <DropdownToggle
                                                aria-haspopup={true}
                                                caret={true}
                                                color="secondary"
                                                nav={true}
                                    >
                                                Account
                                    </DropdownToggle>
                                    <DropdownMenu
                                                flip={true}
                                                right={true}
                                                tag="div"
                                    >
                                                <DropdownItem
                                                            tag={[Function]}
                                                            to="/login"
                                                            toggle={true}
                                                >
                                                            Login
                                                </DropdownItem>
                                    </DropdownMenu>
                        </Unknown>
            </UncontrolledDropdown>
</Nav>,
          "enter": true,
          "exit": true,
          "in": false,
          "isOpen": false,
          "mountOnEnter": false,
          "navbar": true,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 350,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <UncontrolledDropdown
              inNavbar={true}
              nav={true}
>
              <Unknown>
                            <DropdownToggle
                                          aria-haspopup={true}
                                          caret={true}
                                          color="secondary"
                                          nav={true}
                            >
                                          Account
                            </DropdownToggle>
                            <DropdownMenu
                                          flip={true}
                                          right={true}
                                          tag="div"
                            >
                                          <DropdownItem
                                                        tag={[Function]}
                                                        to="/login"
                                                        toggle={true}
                                          >
                                                        Login
                                          </DropdownItem>
                            </DropdownMenu>
              </Unknown>
</UncontrolledDropdown>,
            "className": "ml-auto",
            "navbar": true,
            "tag": "ul",
            "vertical": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Unknown>
                <DropdownToggle
                                aria-haspopup={true}
                                caret={true}
                                color="secondary"
                                nav={true}
                >
                                Account
                </DropdownToggle>
                <DropdownMenu
                                flip={true}
                                right={true}
                                tag="div"
                >
                                <DropdownItem
                                                tag={[Function]}
                                                to="/login"
                                                toggle={true}
                                >
                                                Login
                                </DropdownItem>
                </DropdownMenu>
</Unknown>,
              "inNavbar": true,
              "nav": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <DropdownToggle
                    aria-haspopup={true}
                    caret={true}
                    color="secondary"
                    nav={true}
>
                    Account
</DropdownToggle>,
                  <DropdownMenu
                    flip={true}
                    right={true}
                    tag="div"
>
                    <DropdownItem
                                        tag={[Function]}
                                        to="/login"
                                        toggle={true}
                    >
                                        Login
                    </DropdownItem>
</DropdownMenu>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-haspopup": true,
                    "caret": true,
                    "children": "Account",
                    "color": "secondary",
                    "nav": true,
                  },
                  "ref": null,
                  "rendered": "Account",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <DropdownItem
                      tag={[Function]}
                      to="/login"
                      toggle={true}
>
                      Login
</DropdownItem>,
                    "flip": true,
                    "right": true,
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Login",
                      "tag": [Function],
                      "to": "/login",
                      "toggle": true,
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": Symbol(react.fragment),
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavbarBrand
            tag={[Function]}
            to="/"
>
            famoustitle
</NavbarBrand>,
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
/>,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
>
            <Nav
                        className="ml-auto"
                        navbar={true}
                        tag="ul"
                        vertical={false}
            >
                        <UncontrolledDropdown
                                    inNavbar={true}
                                    nav={true}
                        >
                                    <Unknown>
                                                <DropdownToggle
                                                            aria-haspopup={true}
                                                            caret={true}
                                                            color="secondary"
                                                            nav={true}
                                                >
                                                            Account
                                                </DropdownToggle>
                                                <DropdownMenu
                                                            flip={true}
                                                            right={true}
                                                            tag="div"
                                                >
                                                            <DropdownItem
                                                                        tag={[Function]}
                                                                        to="/login"
                                                                        toggle={true}
                                                            >
                                                                        Login
                                                            </DropdownItem>
                                                </DropdownMenu>
                                    </Unknown>
                        </UncontrolledDropdown>
            </Nav>
</Collapse>,
        ],
        "color": "faded",
        "expand": "md",
        "light": true,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "famoustitle",
            "tag": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "famoustitle",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
>
              <UncontrolledDropdown
                            inNavbar={true}
                            nav={true}
              >
                            <Unknown>
                                          <DropdownToggle
                                                        aria-haspopup={true}
                                                        caret={true}
                                                        color="secondary"
                                                        nav={true}
                                          >
                                                        Account
                                          </DropdownToggle>
                                          <DropdownMenu
                                                        flip={true}
                                                        right={true}
                                                        tag="div"
                                          >
                                                        <DropdownItem
                                                                      tag={[Function]}
                                                                      to="/login"
                                                                      toggle={true}
                                                        >
                                                                      Login
                                                        </DropdownItem>
                                          </DropdownMenu>
                            </Unknown>
              </UncontrolledDropdown>
</Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <UncontrolledDropdown
                inNavbar={true}
                nav={true}
>
                <Unknown>
                                <DropdownToggle
                                                aria-haspopup={true}
                                                caret={true}
                                                color="secondary"
                                                nav={true}
                                >
                                                Account
                                </DropdownToggle>
                                <DropdownMenu
                                                flip={true}
                                                right={true}
                                                tag="div"
                                >
                                                <DropdownItem
                                                                tag={[Function]}
                                                                to="/login"
                                                                toggle={true}
                                                >
                                                                Login
                                                </DropdownItem>
                                </DropdownMenu>
                </Unknown>
</UncontrolledDropdown>,
              "className": "ml-auto",
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown>
                  <DropdownToggle
                                    aria-haspopup={true}
                                    caret={true}
                                    color="secondary"
                                    nav={true}
                  >
                                    Account
                  </DropdownToggle>
                  <DropdownMenu
                                    flip={true}
                                    right={true}
                                    tag="div"
                  >
                                    <DropdownItem
                                                      tag={[Function]}
                                                      to="/login"
                                                      toggle={true}
                                    >
                                                      Login
                                    </DropdownItem>
                  </DropdownMenu>
</Unknown>,
                "inNavbar": true,
                "nav": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <DropdownToggle
                      aria-haspopup={true}
                      caret={true}
                      color="secondary"
                      nav={true}
>
                      Account
</DropdownToggle>,
                    <DropdownMenu
                      flip={true}
                      right={true}
                      tag="div"
>
                      <DropdownItem
                                            tag={[Function]}
                                            to="/login"
                                            toggle={true}
                      >
                                            Login
                      </DropdownItem>
</DropdownMenu>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "aria-haspopup": true,
                      "caret": true,
                      "children": "Account",
                      "color": "secondary",
                      "nav": true,
                    },
                    "ref": null,
                    "rendered": "Account",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <DropdownItem
                        tag={[Function]}
                        to="/login"
                        toggle={true}
>
                        Login
</DropdownItem>,
                      "flip": true,
                      "right": true,
                      "tag": "div",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Login",
                        "tag": [Function],
                        "to": "/login",
                        "toggle": true,
                      },
                      "ref": null,
                      "rendered": "Login",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": Symbol(react.fragment),
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
